src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java

```diff
@@ -2007,6 +2079,46 @@ public void onWakeUp()
         {
             homeBuilding.onWakeUp();
         }
+
+        setIsAsleep(false);
+        dataManager.set(DATA_BED_POS, new BlockPos(0,0,0));
+    }
+
+    public boolean isAsleep()
+    {
+        return dataManager.get(DATA_IS_ASLEEP);
+    }
+
+    public void setIsAsleep(boolean isAsleep)
+    {
+        dataManager.set(DATA_IS_ASLEEP, isAsleep);
+    }
+
+    /**
+     * Returns the orientation of the bed in degrees.
+     */
+    @SideOnly(Side.CLIENT)
+    public float getBedOrientationInDegrees()
+    {
+        IBlockState state = this.getBedLocation() == null ? null : this.world.getBlockState(getBedLocation());
+        if (state != null && state.getBlock().isBed(state, world, getBedLocation(), this))
+        {
+            EnumFacing enumfacing = state.getBlock().getBedDirection(state, world, getBedLocation());
+
+            switch (enumfacing)
+            {
+                case SOUTH:
+                    return 90.0F;
+                case WEST:
+                    return 0.0F;
+                case NORTH:
+                    return 270.0F;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 270.0F to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)